Docs(EuiDataGrid): fix disabled draggable columns examples

mgadewoll opened the pull request:
## Summary

This PR fixes the docs examples for draggable EuiDatagrid columns as they were missing the prop `columnVisibility.canDragAndDropColumns` to be enabled 🤷‍♀️  

This additionally adds the missing draggable columns example to the EUI+ docs.

## QA

- [x] "Draggable Columns" examples in [EUI docs](https://eui.elastic.co/pr_8229/#/tabular-content/data-grid-schema-columns#draggable-columns) and [EUI+ docs](https://eui.elastic.co/pr_8229/new-docs/docs/tabular-content/data-grid/schema-and-columns/#draggable-columns) work as expected (columns can be reordered on drag)

kibanamachine commented:
Preview staging links for this PR:
- Docs site: https://eui.elastic.co/pr_8229/
- Storybook: https://eui.elastic.co/pr_8229/storybook
- New docs: https://eui.elastic.co/pr_8229/new-docs

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3229)
* Commit: 72bff84ce0d5e0230476dfc81d6dd0263ba4146c

<!--buildkite-pr-comment-eui-pull-request-test-and-deploy
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3229","number":3229,"commit":"72bff84ce0d5e0230476dfc81d6dd0263ba4146c"}],"number":3229}
buildkite-pr-comment-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3231)
* Commit: cff1f958261d0471d1c7889b064979ed8078cb6d

### History
* :green_heart: [Build #3229](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3229) succeeded 72bff84ce0d5e0230476dfc81d6dd0263ba4146c

<!--buildkite-pr-comment-eui-pull-request-test-and-deploy
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3231","number":3231,"commit":"cff1f958261d0471d1c7889b064979ed8078cb6d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3229","number":3229,"commit":"72bff84ce0d5e0230476dfc81d6dd0263ba4146c"}],"number":3231}
buildkite-pr-comment-->

weronikaolejniczak commented on packages/website/docs/components/tabular_content/data_grid_schema_and_columns.mdx:
```diff
@@ -364,6 +364,105 @@ export default () => {
 
+## Draggable columns
+
+To reorder columns directly instead of via the actions menu popover, you can enable draggable EuiDataGrid header columns via the `columnVisibility.canDragAndDropColumns` prop. 
```
**Nit-pick**:
```suggestion
To reorder columns directly instead of using the actions menu popover, you can enable draggable EuiDataGrid header columns via the `columnVisibility.canDragAndDropColumns` prop. 
```

weronikaolejniczak commented:
Just a lil nit-pick, I think we could improve the phrasing.

**Tested both in the old and new docs** ✅

mgadewoll commented on packages/website/docs/components/tabular_content/data_grid_schema_and_columns.mdx:
```diff
@@ -364,6 +364,105 @@ export default () => {
+
+To reorder columns directly instead of via the actions menu popover, you can enable draggable EuiDataGrid header columns via the `columnVisibility.canDragAndDropColumns` prop. 
+
+This will allow you to reorder the column by dragging them.
```
Yeah, I just copied there from EUI docs 😄 
Updated for both EUI and EUI+ docs ([commit](https://github.com/elastic/eui/pull/8229/commits/f1d9ac92caa65df4e23af499d474684813e5c96e)).

cee-chen commented:
I also fixed this in https://github.com/elastic/eui/pull/8183, not sure if we'd want to wait for that PR to land. Planning on addressing feedback on it today 🙏

mgadewoll commented:
> I also fixed this in #8183, not sure if we'd want to wait for that PR to land. Planning on addressing feedback on it today 🙏

@cee-chen Ah, I did not realize it was already addressed. I merely noticed it on the DataGrid high contrast mode PR today 😄  I'm fine to close this PR if it's already handled 👍

kibanamachine commented:
Preview staging links for this PR:
- Docs site: https://eui.elastic.co/pr_8229/
- Storybook: https://eui.elastic.co/pr_8229/storybook
- New docs: https://eui.elastic.co/pr_8229/new-docs

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3235)
* Commit: f1d9ac92caa65df4e23af499d474684813e5c96e

### History
* :green_heart: [Build #3231](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3231) succeeded cff1f958261d0471d1c7889b064979ed8078cb6d
* :green_heart: [Build #3229](https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3229) succeeded 72bff84ce0d5e0230476dfc81d6dd0263ba4146c

<!--buildkite-pr-comment-eui-pull-request-test-and-deploy
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3235","number":3235,"commit":"f1d9ac92caa65df4e23af499d474684813e5c96e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3231","number":3231,"commit":"cff1f958261d0471d1c7889b064979ed8078cb6d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/eui-pull-request-test-and-deploy/builds/3229","number":3229,"commit":"72bff84ce0d5e0230476dfc81d6dd0263ba4146c"}],"number":3235}
buildkite-pr-comment-->
